site/component/manifests/wordpress.pp

```diff
@@ -0,0 +1,37 @@
+class component::wordpress (
+  $path             = hiera('path', '/var/www/app_name'),
+  $vhost            = hiera('vhost', 'app-name.dev'),
+  $vhost_port       = 80,
+  $ssl              = false,
+  $mutisite         = false
+) {
+  if $ssl {
+    exec {'create_self_signed_sslcert':
+      command => "openssl req -newkey rsa:2048 -nodes -keyout ${vhost}.key  -x509 -days 365 -out ${vhost}.crt -subj '/CN=${vhost}'",
+      creates => [ "/etc/ssl/${vhost}.key", "/etc/${vhost}.crt", ],
+      path    => ["/usr/bin", "/usr/sbin"],
+      # require => Resource["${vhost}-${vhost_port}-wordpress"]
+    }
+  }
+  ## create vhost for Wordpress
+  nginx::resource::vhost { "${vhost}-${vhost_port}-wordpress" :
+    ensure              => present,
+    server_name         => [$vhost],
+    ssl                 => $ssl,
+    ssl_cert            => "/vagrant/${vhost}.cert",
+    ssl_key             => "/vagrant/${vhost}.key",
```

Comment:
Please make the ssl cert und key configurable through hiera. It would probably make sense to prepend `/vagrant` to the path provided in hiera so that it refers to the source root directory.